### PR TITLE
Fix invisible power levels tile when showing hidden events

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/element-web/issues/23013

image

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Fix invisible power levels tile when showing hidden events ([\#9162](https://github.com/matrix-org/matrix-react-sdk/pull/9162)). Fixes vector-im/element-web#23013.

### CODE REVIEW

File: src/events/EventTileFactory.tsx [hunk 1]
```diff
@@ -225,6 +225,10 @@ export function pickFactory(
             return noEventFactoryFactory(); // improper event type to render
         }
 
+        if (STATE_EVENT_TILE_TYPES[evType] === TextualEventFactory && !hasText(mxEvent, showHiddenEvents)) {
+            return noEventFactoryFactory();
+        }
```
Comment (SimonBrandner) [inline src/events/EventTileFactory.tsx:230]: I am confused - do we just hide the event?

**Review by SimonBrandner — CHANGES_REQUESTED** [review]
Tests please

File: src/events/EventTileFactory.tsx (hunk 1 above)
Comment (t3chguy) [reply src/events/EventTileFactory.tsx:230]: `noEventFactoryFactory` is JSONEventFactory when show hidden events is enabled, see screenshot in the PR description

Comment (SimonBrandner) [reply src/events/EventTileFactory.tsx:230]: NVM, /me was confused, sorry... Do I understand this correctly that you're hiding power-level events that have no diff?

Comment (t3chguy) [reply src/events/EventTileFactory.tsx:230]: That's what already happened, this doesn't change that, events with no text don't render, this just applies that same logic to show hidden events and sticks the json view there

Comment (SimonBrandner) [reply src/events/EventTileFactory.tsx:230]: Makes sense now - _resolved_

**Review by SimonBrandner — APPROVED** [review]

**Pull request merged**